### PR TITLE
feat: Increase product fun by ~25%

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

Adds a little helper to ensure stress levels stay low and overall enjoyment of the product is high. Initial internal user testing reported a **32% increase** in joy, **40% reduction** in stress and **200% increase** in hedgehog related thoughts.

## Changes

Screenshot 2022-10-01 at 15 54 39

* also tidied a couple of Links whilst i was here...
* defaults to off and once enabled in the help bar sticks around until you click hedgey and say Good Bye!

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

Played with the little dude for at least an hour.

### CODE REVIEW

File: frontend/src/lib/components/HedgehogBuddy/HedgehogBuddy.tsx [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+    const [animationName, setAnimationName] = useState('fall')
+    const animation = animations[animationName]
+
+    useEffect(() => {
+        let timer: any = null
+        let iterationsCountdown = animation.maxIteration
+            ? Math.max(1, Math.floor(Math.random() * animation.maxIteration))
+            : null
+
+        console.log(
```
Comment (pauldambra) [inline frontend/src/lib/components/HedgehogBuddy/HedgehogBuddy.tsx:105]: should remove log?

**Review by pauldambra — APPROVED** [review]
:shipit:

**Comment by mariusandra** [comment]
I see the hog flickering on the gifs when the hog switches states. Would image preloading help here?

**Comment by benjackwhite** [comment]
I _think_ I solved the pre-loading issue. Or at least as good as I can for a sensible time invest 😅

**Review by lottiecoxon — APPROVED** [review]
this sparks so much joy - approved!

File: frontend/src/lib/components/HedgehogBuddy/HedgehogBuddy.tsx [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+                    bottom: position.current[1],
+                    transition: !isDragging ? `all ${1000 / fps}ms` : undefined,
+                    transform: `scaleX(${directionRef.current === 'right' ? 1 : -1})`,
+                    cursor: 'pointer',
+                    zIndex: 1001,
+                }}
+            >
+                <div
+                    // eslint-disable-next-line react/forbid-dom-props
+                    style={{
```
[the remark this answers is not in the document]
Comment (benjackwhite) [reply frontend/src/lib/components/HedgehogBuddy/HedgehogBuddy.tsx:234]: That is an awesome tip :o

**Comment by mariusandra** [comment]
The only thing that still doesn't spark maximum joy is this text selection bug:

**Comment by elise-lcdb** [comment]
Is the hedgehog still available on the new UI update? Where can I find it?

**Comment by pauldambra** [comment]
It's in feature previews now "hedgehog mode"

https://us.posthog.com/settings/project#panel=feature-previews

🦔

**Comment by kylealwyn** [comment]
Not seeing it in feature previews :(

**Pull request merged**